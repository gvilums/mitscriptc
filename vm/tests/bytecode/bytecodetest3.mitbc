function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions =
			[
				function
				{
					functions = [],
					constants = [1, 1, 0, 1, 2, 1, 2, 1, 2, 1, 2, None],
					parameter_count = 1,
					local_vars = [val, cur, temp],
					local_ref_vars = [],
					free_vars = [this],
					names = [arr, idx, idx, idx, idx, arr, arr, arr, arr, arr, arr],
					instructions = 
					[
						push_ref	0
						load_ref
						field_load	0
						push_ref	0
						load_ref
						field_load	1
						load_local	0
						index_store
						push_ref	0
						load_ref
						push_ref	0
						load_ref
						field_load	2
						load_const	0
						add
						field_store	3
						push_ref	0
						load_ref
						field_load	4
						load_const	1
						sub
						store_local	1
						load_local	1
						load_const	2
						eq
						not
						push_ref	0
						load_ref
						field_load	5
						load_local	1
						index_load
						push_ref	0
						load_ref
						field_load	6
						load_local	1
						load_const	3
						sub
						load_const	4
						div
						index_load
						geq
						not
						and
						if	2
						goto	38
						push_ref	0
						load_ref
						field_load	7
						load_local	1
						index_load
						store_local	2
						push_ref	0
						load_ref
						field_load	8
						load_local	1
						push_ref	0
						load_ref
						field_load	9
						load_local	1
						load_const	5
						sub
						load_const	6
						div
						index_load
						index_store
						push_ref	0
						load_ref
						field_load	10
						load_local	1
						load_const	7
						sub
						load_const	8
						div
						load_local	2
						index_store
						load_local	1
						load_const	9
						sub
						load_const	10
						div
						store_local	1
						goto	-59
						load_const	11
						return
					]
				},
				function
				{
					functions = [],
					constants = [0, 0, None],
					parameter_count = 0,
					local_vars = [],
					local_ref_vars = [],
					free_vars = [this],
					names = [idx, arr],
					instructions = 
					[
						push_ref	0
						load_ref
						field_load	0
						load_const	0
						gt
						if	2
						goto	7
						push_ref	0
						load_ref
						field_load	1
						load_const	1
						index_load
						return
						load_const	2
						return
					]
				},
				function
				{
					functions = [],
					constants = [0, 0, 1, 0, 2, 1, 2, 1, 2, 1, 2, 2, 2, None, None],
					parameter_count = 0,
					local_vars = [cur, nxt, temp],
					local_ref_vars = [],
					free_vars = [this],
					names = [idx, arr, arr, idx, idx, idx, arr, arr, idx, arr, arr, arr, arr, arr, arr],
					instructions = 
					[
						push_ref	0
						load_ref
						field_load	0
						load_const	0
						gt
						if	2
						goto	125
						push_ref	0
						load_ref
						field_load	1
						load_const	1
						push_ref	0
						load_ref
						field_load	2
						push_ref	0
						load_ref
						field_load	3
						load_const	2
						sub
						index_load
						index_store
						load_const	3
						store_local	0
						load_local	0
						push_ref	0
						load_ref
						field_load	4
						geq
						not
						if	2
						goto	101
						load_local	0
						store_local	1
						load_local	0
						load_const	4
						mul
						load_const	5
						add
						push_ref	0
						load_ref
						field_load	5
						geq
						not
						if	2
						goto	24
						push_ref	0
						load_ref
						field_load	6
						load_local	1
						index_load
						push_ref	0
						load_ref
						field_load	7
						load_local	0
						load_const	6
						mul
						load_const	7
						add
						index_load
						gt
						if	2
						goto	7
						load_local	0
						load_const	8
						mul
						load_const	9
						add
						store_local	1
						load_local	0
						load_const	10
						mul
						push_ref	0
						load_ref
						field_load	8
						geq
						not
						if	2
						goto	20
						push_ref	0
						load_ref
						field_load	9
						load_local	1
						index_load
						push_ref	0
						load_ref
						field_load	10
						load_local	0
						load_const	11
						mul
						index_load
						gt
						if	2
						goto	5
						load_local	0
						load_const	12
						mul
						store_local	1
						load_local	1
						load_local	0
						eq
						not
						if	2
						goto	26
						push_ref	0
						load_ref
						field_load	11
						load_local	0
						index_load
						store_local	2
						push_ref	0
						load_ref
						field_load	12
						load_local	0
						push_ref	0
						load_ref
						field_load	13
						load_local	1
						index_load
						index_store
						push_ref	0
						load_ref
						field_load	14
						load_local	1
						load_local	2
						index_store
						load_local	1
						store_local	0
						goto	3
						load_const	13
						return
						goto	-107
						load_const	14
						return
					]
				}
			],
			constants = [0, None],
			parameter_count = 0,
			local_vars = [this],
			local_ref_vars = [this],
			free_vars = [],
			names = [arr, idx, ins, min, delmin],
			instructions = 
			[
				push_ref	0
				alloc_record
				dup
				alloc_record
				field_store	0
				dup
				load_const	0
				field_store	1
				dup
				load_func	0
				push_ref	0
				alloc_closure	1
				field_store	2
				dup
				load_func	1
				push_ref	0
				alloc_closure	1
				field_store	3
				dup
				load_func	2
				push_ref	0
				alloc_closure	1
				field_store	4
				store_ref
				push_ref	0
				load_ref
				return
				load_const	1
				return
			]
		}
	],
	constants = [0, 2000, 5, 1, 3],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [print, input, intcast, minHeap, h, i, ins, min, delmin, min, delmin, min, ins, min, delmin, min],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	0
		load_func	1
		alloc_closure	0
		store_global	1
		load_func	2
		alloc_closure	0
		store_global	2
		load_func	3
		alloc_closure	0
		store_global	3
		load_global	3
		call	0
		store_global	4
		load_const	0
		store_global	5
		load_global	5
		load_const	1
		geq
		not
		if	2
		goto	12
		load_global	4
		field_load	6
		load_global	5
		load_const	2
		add
		call	1
		load_global	5
		load_const	3
		add
		store_global	5
		goto	-16
		load_global	0
		load_global	4
		field_load	7
		call	0
		call	1
		load_global	4
		field_load	8
		call	0
		load_global	0
		load_global	4
		field_load	9
		call	0
		call	1
		load_global	4
		field_load	10
		call	0
		load_global	0
		load_global	4
		field_load	11
		call	0
		call	1
		load_global	4
		field_load	12
		load_const	4
		call	1
		load_global	0
		load_global	4
		field_load	13
		call	0
		call	1
		load_global	4
		field_load	14
		call	0
		load_global	0
		load_global	4
		field_load	15
		call	0
		call	1
	]
}