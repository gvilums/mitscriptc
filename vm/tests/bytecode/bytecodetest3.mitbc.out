5
6
7
3
8
